Subject: [PATCH] EC handler with no ByteAcc fix                                                                                
From: trenn@suse.de


bug #67364
Delete warning for BIOS not using ByteAcc in Embedded Controller.

Increment offset for Embedded Controller reads with a bit width 
greater than 8.


--- x/drivers/acpi/ec.c.orig	2005-03-07 22:12:19.000000000 +0100
+++ y/drivers/acpi/ec.c	2005-03-07 23:59:52.000000000 +0100
@@ -451,7 +451,7 @@
 		return_VALUE(AE_BAD_PARAMETER);
 
 	if(bit_width != 8) {
-		printk(KERN_WARNING PREFIX "acpi_ec_space_handler: bit_width should be 8\n");
+		/* Embedded Controller region does not use ByteAcc */
 		if (acpi_strict)
 			return_VALUE(AE_BAD_PARAMETER);
 	}
@@ -461,11 +461,11 @@
 next_byte:
 	switch (function) {
 	case ACPI_READ:
-		result = acpi_ec_read(ec, (u8) address, &temp);
+		result = acpi_ec_read(ec, (u8) address + i, &temp);
 		*value = (acpi_integer) temp;
 		break;
 	case ACPI_WRITE:
-		result = acpi_ec_write(ec, (u8) address, (u8) *value);
+		result = acpi_ec_write(ec, (u8) address + i, (u8) *value);
 		break;
 	default:
 		result = -EINVAL;
